mbed_project/_internal/git_utils.py

```diff
@@ -0,0 +1,77 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Wrappers for git operations."""
+from dataclasses import dataclass
+from pathlib import Path
+
+import git
+
+from mbed_project.exceptions import VersionControlError
+from mbed_project._internal.progress import ProgressReporter
+
+
+@dataclass
+class GitReference:
+    """Git reference for a remote repository.
+
+    Attributes:
+        repo_url: URL of the git repository.
+        ref: The reference commit sha, tag or branch.
+    """
+
+    repo_url: str
+    ref: str
+
+
+def clone(url: str, dst_dir: Path) -> git.Repo:
+    """Clone a library repository.
+
+    Args:
+        git_ref: Object containing git information.
+        dst_dir: Destination directory for the cloned repo.
+
+    Raises:
+        VersionControlError: Cloning the repository failed.
+    """
+    try:
+        return git.Repo.clone_from(url, str(dst_dir), progress=ProgressReporter(name=url))
+    except git.exc.GitCommandError as err:
```

Comment:
shouldn't `repo_url` and `ref` be populated after this?

Reply:
Not sure what you mean. They're populated when the .lib file is read. All this function does is perform a git clone.